starwars-characters/src/App.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import Character from './character'
+import './App.css';
+import HomeWorld from './homeworld'; 
+import CharacterList from './characters';
+import { BrowserRouter as Router, Route, Link} from 'react-router-dom';
+
+function Home(){
+return <div><CharacterList /></div>
```

Comment:
These functions are returning components and not doing much else, so its some unecessary nesting. instead have the components in the list render the CharacterList, Homeworld, Character.
Also by doing it this way you are passing route params to these functional components and not the actual page, so they will never be able to read the character id's etc.